src/app/main/component/auth/components/sign-up/sign-up.validator.ts

```diff
@@ -7,6 +7,8 @@ export function ConfirmPasswordValidator(controlName: string, matchingControlNam
     const matchingControl = formGroup.controls[matchingControlName];
     if (matchingControl.value === '') {
       matchingControl.setErrors({ required: true });
+    } else if (control.value === '' && matchingControl.value.length > 0) {
```

Comment:
else if (! control.value && matchingControl.value.length)

Reply:
Added changes according to your recommendation